pkg/conversion/meta.go

```diff
@@ -93,6 +95,10 @@ func UpdateVersionAndKind(baseFields []string, versionField, version, kindField,
 
 	field := v.FieldByName(kindField)
 	if !field.IsValid() {
+		// Types defined in the unversioned package is allowed to not having a kindField.
+		if pkg == "unversioned" {
+			return nil
```

Comment:
Does the test exercise this code path? I'm pretty sure it doesn't-- please add a test.